### PR TITLE
new user-exit for validating before pushing into remote

### PR DESCRIPTION
**Pull request by rayatus**
this is a proposal for #6012

### CODE REVIEW

**Comment by larshp** [comment]
I think currently all exits are handled outside the git library, https://abaplint.app/stats/abapGit/abapGit/package_coupling?folder=/src/git

I'd like to avoid creating a dependency to exits from there, however a step or two up, and it should already be using the exit

**Comment by rayatus** [comment]
moved into src/ui/routing/zcl_abapgit_services_git.clas.abap

File: src/exits/zcl_abapgit_exit.clas.abap [hunk 1]
```diff
@@ -341,4 +341,15 @@ CLASS zcl_abapgit_exit IMPLEMENTATION.
 
   ENDMETHOD.
 
+  METHOD zif_abapgit_exit~validate_before_push.
+
+    IF gi_exit IS NOT INITIAL.
+      gi_exit->validate_before_push(
+          is_comment     = is_comment
+          io_stage       = io_stage
+          io_repo        = io_repo ).
```
Comment (larshp) [inline src/exits/zcl_abapgit_exit.clas.abap:350]: catch and ignore like in the other methods

`CATCH cx_sy_ref_is_initial cx_sy_dyn_call_illegal_method ##NO_HANDLER.`

**Comment by rayatus** [comment]
@larshp, what about now?

File: src/ui/routing/zcl_abapgit_services_git.clas.abap [hunk 2]
```diff
@@ -246,4 +258,13 @@ CLASS zcl_abapgit_services_git IMPLEMENTATION.
...
 
   ENDMETHOD.
+  METHOD validate_before_push.
+
+    zcl_abapgit_exit=>get_instance(  )->validate_before_push(
+        is_comment = is_comment
+        io_stage   = io_stage
+        io_repo    = io_repo ).
+
+  ENDMETHOD.
```
Comment (larshp) [inline src/ui/routing/zcl_abapgit_services_git.clas.abap:268]: this indirection is not needed? just call the exit method directly?

Comment (mbtools) [reply src/ui/routing/zcl_abapgit_services_git.clas.abap]: fixed

**Comment by mbtools** [comment]
lgtm

**Pull request merged**